starknet_py/net/udc_deployer/deployer.py

```diff
@@ -176,3 +183,15 @@ def _get_random_salt() -> int:
 _deployer_serializer = serializer_for_function(
     _deployer_abi.functions["deployContract"]
 )
+
+
+def _is_list_of_ints_or_strings(data: Union[List, dict]) -> bool:
+    """
+    Checks if the given data is a list containing only strings or integers.
+
+    Returns:
+    bool: True if the data is a list of strings or integers, False otherwise.
+    """
+    if isinstance(data, list):
+        return all(isinstance(x, (int, str)) for x in data)
+    return False
```

Comment:
1. I will add it
2. Because the data param is actually [calldata](https://github.com/software-mansion/starknet.py/blob/5f953d0284a158479de3cad11e2dd373b33f10b8/starknet_py/net/udc_deployer/deployer.py#L65), so it can be dict also, then False is returned so this contract cannot be deployed without abi, because calldata is a dict/some kind of different structure and has to be serialized first and for this purpose, abi is required